pkg/reconciler/taskrun/validation_test.go

```diff
@@ -0,0 +1,143 @@
+	Describe("The validatePlatformFormat function", func() {
+
+		When("validating a correctly formatted platform string", func() {
+			DescribeTable("it should not return an error",
+				func(platform string) {
+					err := validatePlatformFormat(platform)
+					Expect(err).ShouldNot(HaveOccurred())
+				},
+				Entry("for linux/amd64", "linux/amd64"),
+				Entry("for linux/s390x", "linux/s390x"),
+			)
+		})
+
+		When("validating a special exception platform string", func() {
+			DescribeTable("it should not return an error",
+				func(platform string) {
+					err := validatePlatformFormat(platform)
+					Expect(err).ShouldNot(HaveOccurred())
+				},
+				Entry("for 'local'", "local"),
+				Entry("for 'localhost'", "localhost"),
+				Entry("for 'linux/x86_64'", "linux/x86_64"),
+			)
+		})
+
+		When("validating a malformed platform string", func() {
+			// A helper variable for the length test to keep the table entry clean.
+			longLabel := strings.Repeat("a", 64)
+
+			DescribeTable("it should return an invalid format error",
+				func(platform string) {
+					err := validatePlatformFormat(platform)
+					Expect(err).Should(MatchError(errInvalidPlatformFormat))
+				},
+				// --- Structural Errors ---
+				Entry("because it has too few parts", "linux"),
+				Entry("because it has too many parts", "linux/amd64/extra"),
+				Entry("because the first part is empty", "/amd64"),
+				Entry("because the second part is empty", "linux/"),
+				Entry("because it's just an empty string", ""),
+				Entry("because it has a trailing slash", "linux/amd64/"),
+				// --- DNS-1035 Label Violations ---
+				Entry("because it contains uppercase characters", "linux/AMD64"),
+				Entry("because it contains an underscore", "linux/amd_64"),
+				Entry("because a part starts with a hyphen", "-linux/amd64"),
+				Entry("because a part ends with a hyphen", "linux/amd64-"),
+				Entry("because a part is longer than 63 chars", fmt.Sprintf("linux/%s", longLabel)),
+			)
+		})
+	})
+
+	// This section tests the combined validation function
+	Describe("The validatePlatform function", func() {
+
+		When("the TaskRun contains a valid platform parameter", func() {
+			DescribeTable("it should return the platform and no error",
+				func(platformValue string) {
+					tr := createTrWithPlatform(&platformValue)
+					platform, err := validatePlatform(tr)
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(platform).Should(Equal(platformValue))
+				},
+				Entry("for a standard platform", "linux/amd64"),
+				Entry("for the 'linux/x86_64' exception", "linux/x86_64"),
+			)
+		})
+
+		When("the TaskRun contains an invalid or missing platform parameter", func() {
+			DescribeTable("it should return the corresponding error",
+				func(platformValue *string, expectedError error) {
+					tr := createTrWithPlatform(platformValue)
+					_, err := validatePlatform(tr)
+					Expect(err).Should(MatchError(expectedError))
```

Comment:
```suggestion
					Expect(validatePlatform(tr)).Error().To(MatchError(expectedError))
```

This can be applied to a few other places in this PR.